HoaryFox/Member/CreateTag.cs

```diff
@@ -14,134 +15,166 @@ namespace HoaryFox.Member
+            int endIndex = _nodes.Id.IndexOf(frame.IdNodeEnd[eNum]);
+            var nodeStart = new Point3d(_nodes.X[startIndex], _nodes.Y[startIndex], _nodes.Z[startIndex]);
+            var nodeEnd = new Point3d(_nodes.X[endIndex], _nodes.Y[endIndex], _nodes.Z[endIndex]);
+            Position.Add(new Point3d((nodeStart.X + nodeEnd.X) / 2.0, (nodeStart.Y + nodeEnd.Y) / 2.0, (nodeStart.Z + nodeEnd.Z) / 2.0));
+        }
+
+        private TagInfo TagSteel(StbFrame frame, int idSection)
+        {
+            int idShape;
+            string shapeName;
+            switch (frame.FrameType)
+            {
+                case FrameType.Column:
+                case FrameType.Post:
+                    idShape = _colS.Id.IndexOf(idSection);
+                    shapeName = _colS.Shape[idShape];
+                    break;
+                case FrameType.Girder:
+                case FrameType.Beam:
+                    idShape = _beamS.Id.IndexOf(idSection);
+                    shapeName = _beamS.Shape[idShape];
+                    break;
+                case FrameType.Brace:
+                    idShape = _braceS.Id.IndexOf(idSection);
+                    shapeName = _braceS.Shape[idShape];
+                    break;
+                case FrameType.Slab:
+                case FrameType.Wall:
+                case FrameType.Any:
+                    throw new ArgumentException("Wrong frame type");
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            int secIndex = _secSteel.Name.IndexOf(shapeName);
+            var tagInfo = new TagInfo
+            {
+                Name = _secSteel.Name[secIndex],
+                ShapeTypes = _secSteel.ShapeType[secIndex],
+                P1 = _secSteel.P1[secIndex],
+                P2 = _secSteel.P2[secIndex],
+                P3 = _secSteel.P3[secIndex],
+                P4 = _secSteel.P4[secIndex]
+            };
+            return tagInfo;
+        }
+
+        private TagInfo TagRc(StbFrame frame, int idSection)
+        {
+            int secIndex;
+            TagInfo tagInfo;
+            switch (frame.FrameType)
+            {
+                case FrameType.Column:
+                case FrameType.Post:
+                    secIndex = _colRc.Id.IndexOf(idSection);
+                    tagInfo = new TagInfo(_colRc.Name[secIndex], _colRc.Height[secIndex], _colRc.Width[secIndex], 0d, 0d);
+                    break;
+                case FrameType.Girder:
+                case FrameType.Beam:
+                    secIndex = _beamRc.Id.IndexOf(idSection);
+                    tagInfo = new TagInfo(_beamRc.Name[secIndex], _beamRc.Depth[secIndex], _beamRc.Width[secIndex], 0d, 0d);
+                    break;
+                case FrameType.Brace:
+                case FrameType.Slab:
+                case FrameType.Wall:
+                case FrameType.Any:
+                    throw new Exception("Wrong frame type");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/hrntsm/HoaryFox/pullRequest?prid=6419994)